bin/varnishd/cache/cache_range.c

```diff
@@ -180,13 +181,66 @@ vrg_dorange(struct req *req, const char *r, void **priv)
 	return (NULL);
 }
 
+/*
+ * return 1 if range should be observed, based on if-range value
+ * if-range can either be a date or an ETag [RFC7233 3.2 p8]
+ */
+static int
+vrg_ifrange(struct req *req)
+{
+	const char *p, *e;
+	vtim_real ims, lm, d;
+
+	if (!http_GetHdr(req->http, H_If_Range, &p))
+		return (1);
+
+	/* strong validation needed */
+	if (p[0] == 'W' && p[1] == '/')
+		return (0);
+
+	/* ETag */
+	if (p[0] == '"') {
+		if (!http_GetHdr(req->resp, H_ETag, &e))
+			return (0);
+		if ((e[0] == 'W' && e[1] == '/'))
+			return (0);
```

Comment:
if the etag is weak, we should fall through to LM comparison, rather than returning.

Reply:
Ah no, I was wrong, the If-Range explicitly demands an etag, so there is no fallback. Sorry.